### PR TITLE
FormBadge fix

### LINKED ISSUE #214: The green box is incorrectly profiled in the token section
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/aleks/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/aleks/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">

</head>

<body link="#0563C1" vlink="#954F72">

Description | Steps | Expected Results | Current results
-- | -- | -- | --
Add   Wallet | 1) Open dashboard     2) Go to the wallets tab     3) Click "Add New Wallet"     4) Choose Type Executable     5) Complete wallet name and use correct wallet address     6) Click "proceed" and select tokens, and click "proceed"     7) in the selected tokens click ‘max’ | The green box is correctly   profiled to the right in the token section and the numbers do not enter the   ‘true’ sign | The green box is incorrectly   profiled in the token section and the numbers in the box overlap the ‘true’   sign

</body>

</html>

AddWallet1

### PR DESCRIPTION
**Pull request by anhelinakruk**
When adding a new wallet or transferring tokens, there is a modal to select tokens and then set approval limits. FormBadge - it's a row with: token + checkbox and token + approval limit input.

There were a lot of problems with the styles, so I decided to do it from 0. Then I implemented it into the app and deleted the old version.

Before:
Screenshot 2024-06-27 at 10 26 02 Screenshot 2024-06-27 at 10 26 41

After: 
Screenshot 2024-06-27 at 10 29 42 
Screenshot 2024-06-27 at 10 30 04.

The same with the transfer modal.

### CODE REVIEW

File: src/components/FormBadge/FormBadge.stories.tsx [hunk 1]
```diff
@@ -1,43 +1,55 @@
-import type { Meta, StoryObj } from "storybook-framework-qwik";
-import { FormBadge, type FormBadgeProps } from "./FormBadge";
-import { CheckBox } from "../Checkbox/Checkbox";
-import { Input } from "../Input/Input";
 import { $ } from "@builder.io/qwik";
+import Checkbox from "../Atoms/Checkbox/Checkbox";
+import Input from "../Atoms/Input/Input";
+import { FormBadge, type FormBadgeProps } from "./FormBadge";
+import IconSuccess from "@material-design-icons/svg/round/check_circle_outline.svg?jsx";
```
Comment (Uacias) [inline src/components/FormBadge/FormBadge.stories.tsx]: Defined and never used I guess

File: src/components/FormBadge/FormBadge.stories.tsx [hunk 2]
```diff
@@ -1,43 +1,55 @@
...
+  );
+}
+
+export function ApprovalLimit(args: FormBadgeProps) {
+  const onInput = $((e: any) => {
+    const target = e.target;
+    const value = target.value;
+    console.log(value);
+    return value;
+  });
```
Comment (Uacias) [inline src/components/FormBadge/FormBadge.stories.tsx]: Defined and never used, btw remove console.log

File: src/components/FormBadge/FormBadge.stories.tsx [hunk 3]
```diff
@@ -1,43 +1,55 @@
...
+
+  const isValid = $((value: string) => {
```
Comment (Uacias) [inline src/components/FormBadge/FormBadge.stories.tsx]: Defined and never used

File: src/components/FormBadge/FormBadge.stories.tsx [hunk 4]
```diff
@@ -1,43 +1,55 @@
...
+    );
+  });
 
-export default meta;
+  return (
+    <FormBadge {...args}>
+      <Input
+        id=""
+        isValid={true}
+        // iconRight={isValid function ? <IconSuccess class="h-4 w-4" /> : null}
```
Comment (Uacias) [inline src/components/FormBadge/FormBadge.stories.tsx]: Do we need that?

File: src/components/Forms/portfolioTransfters/CoinsToTransfer.tsx [hunk 5]
```diff
@@ -45,27 +46,20 @@ export default component$<CoinsToTransferProps>(
...
-                                onClick$={() => {
+                                isChecked={currentCoin?.isChecked ?? false}
+                                onClick={$(() => {
                                   if (currentCoin) {
                                     currentCoin.isChecked =
                                       !currentCoin.isChecked;
                                   }
-                                }}
+                                })}
+                                // value={`${structure.structure.name}${balance.balance.symbol}`}
```
Comment (Uacias) [inline src/components/Forms/portfolioTransfters/CoinsToTransfer.tsx:62]: Do we need that?

File: src/routes/app/wallets/_components/AmountOfCoins.tsx [hunk 6]
```diff
@@ -17,84 +19,101 @@ export default component$<AmountOfCoinsProps>(
...
+            );
+
+            const isValid = () => {
+              return (
+                checkPattern(coin!.amount, /^\d*\.?\d*$/) &&
+                coin!.amount != "0" &&
+                coin!.amount.length > 0 &&
+                coin!.amount[0] != "0"
+              );
+            };
```
Comment (Uacias) [inline src/routes/app/wallets/_components/AmountOfCoins.tsx]: I'd rather use simple if condition to check if coin and coint.amount is not null e.g.
```
if(coin && coin.amount) {
rerturn (...
```
In this case we dont have to use 
```
coin!.amount
``` 
anywhere.
IMO thast's more readable but you can ask other developers for opinion e.g. @pgajek

**Review by Uacias — CHANGES_REQUESTED** [review]
I found a few small things we could improve, take a look and maybe discuss with @pgajek

File: src/components/Forms/portfolioTransfters/CoinsToTransfer.tsx (hunk 5 above)
Comment (anhelinakruk) [reply src/components/Forms/portfolioTransfters/CoinsToTransfer.tsx:62]: That wasn't my comment, so I left it.

File: src/routes/app/wallets/_components/AmountOfCoins.tsx (hunk 6 above)
Comment (anhelinakruk) [reply src/routes/app/wallets/_components/AmountOfCoins.tsx]: Fixed

File: src/components/FormBadge/FormBadge.stories.tsx (hunk 4 above)
Comment (anhelinakruk) [reply src/components/FormBadge/FormBadge.stories.tsx]: Deleted

File: src/components/FormBadge/FormBadge.stories.tsx (hunk 3 above)
Comment (anhelinakruk) [reply src/components/FormBadge/FormBadge.stories.tsx]: Deleted

File: src/components/FormBadge/FormBadge.stories.tsx (hunk 2 above)
Comment (anhelinakruk) [reply src/components/FormBadge/FormBadge.stories.tsx]: Deleted

File: src/components/FormBadge/FormBadge.stories.tsx (hunk 1 above)
Comment (anhelinakruk) [reply src/components/FormBadge/FormBadge.stories.tsx]: Deleted

**Pull request merged**